clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -7485,6 +7485,99 @@ void CodeGenModule::printPostfixForExternalizedDecl(llvm::raw_ostream &OS,
   }
 }
 
+namespace {
+/// A 'teams loop' with a nested 'loop bind(parallel)' or generic function
+/// call in the associated loop-nest cannot be a 'parllel for'.
+class TeamsLoopChecker final : public ConstStmtVisitor<TeamsLoopChecker> {
```

Comment:
Better to do this analysis in Sema and mark the region appropriately

Reply:
Alexey - thanks for the review comments. Good suggestion here. I'll look at moving the analysis.